Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -499,10 +499,17 @@ public struct Reducer<State, Action, Environment> {
         }
         return .none
       }
-      defer { globalState = toLocalState.embed(localState) }
+      
+      var wrappedState = StateWrapper(localState)
+      
+      defer {
+        if wrappedState.stateDidChange {
+          globalState = toLocalState.embed(wrappedState.state)
+        }
+      }
 
       let effects = self.run(
-        &localState,
+        &wrappedState.state,
```

Comment:
@tgrapperon I'm not quite I understand how this is intended to work. Shouldn't this line always set `stateDidChange` to true, thus updating `globalState`? Or do you have provide an example (or better, test case) that demonstrates how `stateDidChange` can remain false?

Reply:
TIL inout arguments are reassigned at the end of the function execution.
https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID545

This is quite a surprise to me, as I was expecting some kind of reference semantic, but it makes sense now. 
I don't think the optimisation they're mentioning kick's in in reducers. Maybe that's something to explore.

I'll need to adjust my mental model though.
All of this makes the PR irrelevant and I'll close it. 
Thanks for reviewing it.